### ISSUE TITLE
Runtime editing

### ISSUE THREAD

**Issue by csibesz07** [rubensousa/BottomSheetBuilder#4]
Hi, i would like to edit, add, remove menu items, submenus without creating a new view with the builder, dynamically. I think a class which allows editing `List<BottomSheetItem> items` would solve it. The recyclerview could animate changes. What do you think? Is it possible?

**Comment by rubensousa** [comment]
Can you explain your use-case? I don't see how editing items in real-time might be a good option

**Comment by csibesz07** [comment]
I'm making a plugin for an app called Tasker. The plugin allows the user to create material style menus, messages. And i thought i would make a preview which changes on user interaction to represent the menu that will appear. I'm new to android but i've learnt a lot making the plugin.

**Comment by rubensousa** [comment]
I see. Do you want to give this a try and send a pull request later? It's not on my plans to implement such behavior, so, any contributions would be helpful.

**Comment by csibesz07** [comment]
Alright, I'll see what I can do.

**Comment by csibesz07** [comment]
Hi, could you please review my changes. After you closed the PR i worked hard to make it simpler and modifiable. I understand that the code was messy and hard to follow. I had to because android menu limits run-time changes and cannot store plus info. So i added my own menu which can store plus infos about items (colors,images,font etc), so it's easier to add functionality to BottomSheetItem and also supports android's menu. Performance wise there shouldn't be difference (only creating BottomSheetMenu which is just a quick menu tree traversal) since run-time structures won't be referenced in case you don't enable run-time editing. I won't bother you again if you don't like the changes i made. I also included sample for dynamic menu in your sample program. Many thanks for considering my request.

**Comment by ch-muhammad-adil** [comment]
@rubensousa I hope I will add this feature, as it is really required. :)

**Comment by ch-muhammad-adil** [comment]
This has been resolved in commit #13 . Sooner release will be added. 

Here is some code to handle dynamic behaviour. 


`private void createShareMenu() {
        BottomSheetBuilder builder = new BottomSheetBuilder(MyCardsActivity.this, R.style.AppTheme_BottomSheetDialog_Custom)
                .setMode(BottomSheetBuilder.MODE_GRID)
                .setItemClickListener(new BottomSheetItemClickListener() {
                    @Override
                    public void onBottomSheetItemClick(MenuItem item) {

                    }
                });
        for (YourMenuItem yourMenuItem : ListOfYourMenuItems) {
            builder.addItem(yourMenuItem.getMenuId(), yourMenuItem.getName(), new BitmapDrawable(getResources(), yourMenuItem.getBitmap()));
        }
        BottomSheetMenuDialog dialog = builder.createDialog();
        dialog.show();
    }`

you need to preload your menu items with bitmap in this case. you can also just pass its drawable.
